### PR TITLE
chore: switch to eslint-config-prettier from eslint-plugin-prettier

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
~Closes #~ [I was told to just go nuts 🙌](https://discord.com/channels/867764511159091230/980118117676118067/1056705880726446130)

## 🎯 Changes

Switches to 🌟 [`eslint-config-prettier`](https://github.com/prettier/eslint-config-prettier) from 🐌 [`eslint-plugin-prettier`](https://github.com/prettier/eslint-plugin-prettier).

- The 🌟 _config_ disables all common Prettier rules that would conflict with Prettier.
- The 🐌 _plugin_ runs Prettier as an ESLint rule named `prettier/prettier` (the `prettier` plugin > its `prettier` rule). Rule reports are seen through running ESLint, such as via the user's editor with the ESLint extension, or via the ESLint CLI.

Since ESLint is no longer running Prettier & providing auto-fixing, this adds a few entry points for Prettier:

- [Git commit hook](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks): [Husky](https://typicode.github.io/husky) sets up a Git precommit hook on pnpm install (`pnpm prepare`) -> [lint-staged](https://github.com/okonet/lint-staged) provides the names of staged files to Prettier. The net result is unless you `--no-verify` your Git commits will have Prettier run on all files.
- [VS Code Workspace Settings](https://code.visualstudio.com/docs/getstarted/settings#_workspace-settings): Sets the default formatter in VS Code to Prettier. Autoformatting on save was already enabled in #416.
- If neither of those make you happy, `pnpm run format:write` works too.

This comes with two main changes _(I'd say improvements!)_ for developers:

- Formatting issues -most commonly, whitespace issues- no longer have red squigglies. I see that as a positive: they won't get in the way of actual ESLint/TypeScript complaints anymore!
- Prettier will run on all files, not just ones linted by ESLint. _(hence the 70+ additional files reformatted in this PR)_

| Before                                                                                                             | After                                                                                                              |
| ------------------------------------------------------------------------------------------------------------------ | ------------------------------------------------------------------------------------------------------------------ |
| <video src="https://user-images.githubusercontent.com/3335181/209631964-65ece2b1-4a70-43bd-a7b4-2a876329639b.mov"> | <video src="https://user-images.githubusercontent.com/3335181/209631992-9499b327-2f67-4d75-9447-f6a8a88e9998.mov"> |

Suggested additional reading: [typescript-eslint's _What About Formatting?_ article](https://typescript-eslint.io/linting/troubleshooting/formatting#suggested-usage). I'm generally willing to die on the hill that formatting should be run separately from the linter... but this is not my repo, and I won't be offended if this PR gets closed with prejudice. 😄

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [x] If necessary, I have added documentation related to the changes made.
- [x] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+npx lint-staged
```
Comment (juliusmarminge) [inline .husky/pre-commit:4]: 🥶

**Review by juliusmarminge — COMMENTED** [review]
My thoughts:

Pros:
- Good thing that we format all files and not just the linted ones.

Cons:
- Not the biggest fan of precommit hooks, I think we've had this brought up before too. Causes slow commits and generally stop me from committing a WIP.

Others:
- Not sure if I like or dislike squiggles due to invalid formatting 🤔
  - Is there not a setting in eslint where you can "turn down" the warning level to make it not show the yellow/red squiggly, if that's the path we wanna go down.

**Comment by JoshuaKGoldberg** [comment]
> Not the biggest fan of precommit hooks, I think we've had this brought up before too. Causes slow commits and generally stop me from committing a WIP.

Yeah that is a downside. FWIW _most_ of the time, when I've experienced or otherwise seen precommit hook pain, it's been from running meatier operations than formatting: most commonly linting or even tests (yikes!).

**Comment by JoshuaKGoldberg** [comment]
> Not the biggest fan of precommit hooks, I think we've had this brought up before too. Causes slow commits and generally stop me from committing a WIP.

Yeah that is a downside. FWIW _most_ of the time, when I've experienced or otherwise seen precommit hook pain, it's been from running meatier operations than formatting: most commonly linting or even tests (yikes!).

If a second or so for commits is still too much, [dprint](https://dprint.dev) is pretty blazingly fast. I'd be up for investigating switching over if you're interested.

> Is there not a setting in eslint where you can "turn down" the warning level to make it not show the yellow/red squiggly, if that's the path we wanna go down.

Yeah! A couple options:

* `"prettier/prettier": "warn"` in the ESLint config switches it from red squigglies to yellow. This would make `--max-warnings 0` actually useful!
* The VS Code ESLint extension has an `eslint.rules.customizations` option that can manually override rule severities. `[{ "rule": "prettier/prettier", "severity": "info" }]`

File: www/docs/server/context.md [hunk 2]
```diff
@@ -18,29 +18,28 @@ The result is propagated to all resolvers.
 // @filename: context.ts
 // -------------------------------------------------
 import type { inferAsyncReturnType } from '@trpc/server';
+// -------------------------------------------------
+// @filename: trpc.ts
+// -------------------------------------------------
+import { TRPCError, initTRPC } from '@trpc/server';
```
[the remark this answers is not in the document]
Comment (JoshuaKGoldberg) [reply www/docs/server/context.md:24]: Ah this is an interesting one, thanks for the catch! The "proper" solution would be to split these up into two code blocks and have some kind of _"look at the other block for types"_ directive in shiki-land (https://github.com/shikijs/twoslash/tree/main/packages/remark-shiki-twoslash#twoslash-includes maybe?).

But I haven't played with shiki-twoslash to feel competent with those. `prettier-ignore` it is.

**Review by KATT — CHANGES_REQUESTED** [review]
Love everything apart from the precommit hooks, please remove them 😅 I want editors to be the place where formatting happens, not CI hooks that tries to block you from doing stuff like non-working draft PRs

**Review by sachinraja — APPROVED** [review]

**Pull request merged**